src/kv/direct.rs

```diff
@@ -0,0 +1,141 @@
+use std::collections::HashMap;
+use std::ops::Deref;
+
+use async_nats::{jetstream::kv::Store, Client};
+use futures::{TryFutureExt, TryStreamExt};
+use serde::de::DeserializeOwned;
+use tracing::debug;
+
+use super::{
+    delete_link, ld_hash_raw, put_link, KvStore, CLAIMS_PREFIX, LINKDEF_PREFIX, SUBJECT_KEY,
+};
+use crate::{types::LinkDefinition, Result};
+
+#[derive(Clone, Debug)]
+pub struct DirectKvStore {
+    store: Store,
+}
+
+impl AsRef<Store> for DirectKvStore {
+    fn as_ref(&self) -> &Store {
+        &self.store
+    }
+}
+
+impl Deref for DirectKvStore {
+    type Target = Store;
+
+    fn deref(&self) -> &Self::Target {
+        &self.store
+    }
+}
+
+impl DirectKvStore {
+    pub async fn new(nc: Client, lattice_prefix: &str, js_domain: Option<String>) -> Result<Self> {
+        super::get_kv_store(nc, lattice_prefix, js_domain)
+            .await
+            .map(|store| Self { store })
+    }
+
+    async fn fetch_data<T: DeserializeOwned>(&self, filter_prefix: &str) -> Result<Vec<T>> {
```

Comment:
I could use some comments/type hints to help follow this logic. If we can't get or parse an entry, does the whole operation fail?